framework/auth/campaigns.py

```diff
@@ -9,7 +9,7 @@
 CAMPAIGNS = ImmutableDict({
     'prereg': {
         'system_tag': 'prereg_challenge_campaign',
-        'redirect_url': lambda: web_url_for('prereg_landing_page'),
+        'redirect_url': lambda: web_url_for('prereg_landing_page', _absolute=True),
```

Comment:
[To CR] When user visits "prereg" not logged in or no account:
Before: `/login?campaign=prereg` -> CAS service url is `https://osf.io/login?next=/prereg/`
Now: `/register?campaign=prereg` -> CAS service url is `https://osf.io/prereg`